code/modules/reagents/chemistry/reagents/toxin_reagents.dm

```diff
@@ -967,3 +967,72 @@
 				to_chat(M, "<span class='warning'>Your missing arm aches from wherever you left it.</span>")
 				M.emote("sigh")
 	return ..()
+
+/datum/reagent/toxin/potash
+	name = "Potash"
+	id = "potash"
+	description = "An odorless white crystal resembling and tasting of salt. Used to cause heart arrhythmias in high doses or cardiac arrest through injection. Also an effective fertiliser."
+	taste_description = "salt"
+	reagent_state = SOLID
+	color = "#FFFFFF" // rgb: 255,255,255
+	toxpwr = 0
+
+/datum/reagent/toxin/potash/reaction_mob(mob/living/carbon/M, method=TOUCH, reac_volume)
+	if (method==INJECT)
+		M.reagents.add_reagent("injectedpotash", reac_volume)
```

Comment:
Haven't done HW but can you not just add potassium chloride and remove the source reagent here before you finish the reaction?

Reply:
No you can't unfortunately, reaction code is run completely separately (and prior) to actually adding the chemical to the mob. At this point in the code the potash isn't actually in the mob yet

Reply:
so even better? you can add the kcl and return so the potash isn't added at all?

Reply:
Uh no, it's basically run like callChemicalReacting(), callChemicalGoIntoMob(), returning from reacting doesnt stop the next line from executing

Reply:
Here went and found actual code. Look in syringes.dm

```
reagents.reaction(L, INJECT, fraction)
reagents.trans_to(target, amount_per_transfer_from_this, transfered_by = user)
```

Reply:
The problem isn't adding the heart attach chem, it's removing the injected chem. So unless the reaction and trans_to are swapped around in syringes.dm (and everywhere else for consistency) there isn't much I can do apart from having a buffer chem. I don't know what kind of affect swapping the two would have but maybe it's a possibility

Reply:
This will not be merged with the intermediate chem unless someone else okays it. Instead of trying to bandaid it may as well fix the foundation to support this interaction.

Reply:
I'll have a go at it